controllers/reconcile.go

```diff
@@ -853,6 +853,94 @@ func setupFedModules(feEnv *crd.FrontendEnvironment, frontendList *crd.FrontendL
 	return nil
 }
 
+func adjustSearchEntry(searchEntry *crd.SearchEntry, frontend crd.Frontend) crd.SearchEntry {
+	newSearchEntry := crd.SearchEntry{
+		// make the id environment and frontend specific to reduce duplicate ids across Frontend resources
+		ID:          fmt.Sprintf("%s-%s-%s", frontend.Name, frontend.Spec.EnvName, searchEntry.ID),
+		Title:       searchEntry.Title,
+		Description: searchEntry.Description,
+		Href:        searchEntry.Href,
+		AltTitle:    searchEntry.AltTitle,
+		IsExternal:  searchEntry.IsExternal,
+	}
+	return newSearchEntry
+}
+
+func setupSearchIndex(feList *crd.FrontendList) []crd.SearchEntry {
+	searchIndex := []crd.SearchEntry{}
+	for _, frontend := range feList.Items {
+		if frontend.Spec.SearchEntries != nil {
+			for _, searchEntry := range frontend.Spec.SearchEntries {
+				searchIndex = append(searchIndex, adjustSearchEntry(searchEntry, frontend))
+			}
+		}
+	}
+	return searchIndex
+}
+
+func setupWidgetRegistry(felist *crd.FrontendList) []crd.WidgetEntry {
+	widgetRegistry := []crd.WidgetEntry{}
+	for _, frontend := range felist.Items {
+		for _, widget := range frontend.Spec.WidgetRegistry {
+			widgetRegistry = append(widgetRegistry, *widget)
+		}
+	}
+
+	return widgetRegistry
+}
+
+func getServiceTilePath(section string, group string) string {
+	return fmt.Sprintf("%s-%s", section, group)
+}
+
+func setupServiceTilesData(felist *crd.FrontendList, feEnvironment crd.FrontendEnvironment) []crd.FrontendServiceCategoryGenerated {
+	categories := []crd.FrontendServiceCategoryGenerated{}
+	if feEnvironment.Spec.ServiceCategories == nil {
+		// skip if we do not have service categories
+		return categories
+	}
+
+	// just a quick cache to make it easier and faster to assign tiles to their destination
+	tileGroupAccessMap := make(map[string]*[]crd.ServiceTile)
+
+	for _, category := range *feEnvironment.Spec.ServiceCategories {
+		groups := []crd.FrontendServiceCategoryGroupGenerated{}
+		for _, gr := range category.Groups {
+			tiles := []crd.ServiceTile{}
+			group := crd.FrontendServiceCategoryGroupGenerated{
+				ID:    gr.ID,
+				Title: gr.Title,
+				Tiles: &tiles,
+			}
+			groups = append(groups, group)
+			groupKey := getServiceTilePath(category.ID, gr.ID)
+			tileGroupAccessMap[groupKey] = &tiles
+		}
+		newCategory := crd.FrontendServiceCategoryGenerated{
+			ID:     category.ID,
+			Title:  category.Title,
+			Groups: groups,
+		}
+
+		categories = append(categories, newCategory)
+	}
+
+	for _, frontend := range felist.Items {
+		if frontend.Spec.ServiceTiles != nil {
+			for _, tile := range frontend.Spec.ServiceTiles {
+				groupKey := getServiceTilePath(tile.Section, tile.Group)
+				// ignore the tile if destination does not exist
```

Comment:
@Hyperkid123 do you think it's worth logging something if the destination doesn't exist?

Reply:
Its definitely not going to hurt.